test/Ocelot.AcceptanceTests/ServiceDiscovery/KubernetesServiceDiscoveryTests.cs

```diff
@@ -311,15 +309,24 @@ private void GivenThereIsAFakeKubernetesProvider(EndpointsV1 endpoints, bool isS
     }
 
     private void WithKubernetes(IServiceCollection services) => services
-        .AddOcelot().AddKubernetes()
-        .Services.RemoveAll<IKubeApiClient>().AddSingleton(_clientFactory);
+        .AddOcelot().AddKubernetes(false)
+        .Services.Configure(_kubeClientOptionsConfigure)
+        .Replace(GetValidateScopesDescriptor());
 
     private void WithKubernetesAndRoundRobin(IServiceCollection services) => services
-        .AddOcelot().AddKubernetes()
+        .AddOcelot().AddKubernetes(false)
```

Comment:
Seems we can reuse `WithKubernetes` at the beginning of the method.
After that replace the rest of services.
```suggestion
        WithKubernetes(services);
        services.BlaBlaReplacements()...
```